src/main/java/org/joshsim/lang/interpret/visitor/JoshConfigParserVisitor.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * ANTLR visitor for parsing Josh configuration files (.jshc format).
+ *
+ * @license BSD-3-Clause
+ */
+
+package org.joshsim.lang.interpret.visitor;
+
+import java.math.BigDecimal;
+import org.joshsim.engine.config.ConfigBuilder;
+import org.joshsim.engine.value.converter.Units;
+import org.joshsim.engine.value.engine.EngineValueFactory;
+import org.joshsim.engine.value.type.EngineValue;
+import org.joshsim.lang.antlr.JoshConfigBaseVisitor;
+import org.joshsim.lang.antlr.JoshConfigParser;
+import org.joshsim.lang.interpret.fragment.jshc.JshcConfigBuilderFragment;
+import org.joshsim.lang.interpret.fragment.jshc.JshcFragment;
+import org.joshsim.lang.interpret.fragment.jshc.JshcValueFragment;
+
+/**
+ * ANTLR visitor for parsing .jshc configuration files.
+ *
+ * <p>This visitor processes ANTLR parse trees generated from .jshc files
+ * and produces JshcFragment instances containing the parsed configuration data.</p>
+ */
+public class JoshConfigParserVisitor extends JoshConfigBaseVisitor<JshcFragment> {
+
+  private final EngineValueFactory valueFactory;
+
+  /**
+   * Creates a new visitor with the specified value factory.
+   *
+   * @param valueFactory The factory to use for creating EngineValues
+   */
+  public JoshConfigParserVisitor(EngineValueFactory valueFactory) {
+    this.valueFactory = valueFactory;
+  }
+
+  /**
+   * Visits the root config context and processes all configuration lines.
+   *
+   * @param ctx The ANTLR config context
+   * @return A JshcFragment containing the complete configuration
+   */
+  @Override
+  public JshcFragment visitConfig(JoshConfigParser.ConfigContext ctx) {
+    ConfigBuilder builder = new ConfigBuilder();
+
+    // Process each config line
+    for (JoshConfigParser.ConfigLineContext lineContext : ctx.configLine()) {
+      if (lineContext.assignment() != null) {
```

Comment:
Humm is there a way for lineContext to use `accept` on `this` as a visitor to avoid this kind of inspection? Perhaps `JshcFragment` could have an `getIsComment` method that, if true, we can ignore? We should do a web search and look at other ANTLR visitors in the project to see if there are other options. If we go this route, perhaps we add a `JshcCommentFragment` subclass.